File: 1-two-sum/2.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const map = new Map()
    for (let [i, a] of nums.entries()) {
        const b = target - a
        if (map.has(b)) return [map.get(b), i]
        map.set(a, i)
    }
    return []
};

File: 1-two-sum/2.cc
#include <iostream>
#include <vector>
#include <unordered_map>

using namespace std;

class Solution {
   public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> map;
        for (int i = 0; i < nums.size(); ++i) {
            int a = nums[i];
            int b = target - nums[i];
            if (map.count(b)) return {map[b], i};
            map[b] = i;
        }
        return {};
    }
};
